pheget/static/js/searchbox.js

```diff
@@ -1,60 +1,113 @@
-var getOmni = function (omniurl, callback) {
-    var xhr = new XMLHttpRequest();
-    xhr.open('GET', omniurl, true);
-    xhr.responseType = 'json';
-    xhr.onload = function () {
-        var status = xhr.status;
-        if (status === 200) {
-            callback(null, xhr.response);
-        } else {
-            callback(status, xhr.response);
-        }
-    };
-    xhr.send();
-};
+// Generic failure callback function
+function failureCallback(error) {
+    console.log('Error status: ' + error);
+}
 
-// eslint-disable-next-line no-unused-vars
-function parseSearch(searchText) {
+
+// Returns the data from an omnisearch fetch if successful, return nothing otherwise
+function getOmniSearch(searchText) {
+    const omniurl = `https://portaldev.sph.umich.edu/api/v1/annotation/omnisearch/?q=${searchText}&build=GRCh38`;
+    var data = fetch(omniurl)
+        .then((response) => response.json())
+        .then((myJson) => myJson.data[0])
+        .catch(function(error) {
+            console.log('Omnisearch error, status code: ' + error);
+            return;
+        });
+    return data;
+}
+
+
+// Returns the data from our internal best variant API by searching for the gene_id (ENSG...) or gene_name
+function getBestVar(symbol) {
+    const bestVarURL = `/api/bestvar/${symbol}`;
+    var data = fetch(bestVarURL)
+        .then((response) => response.json())
```

Comment:
As noted on slack, you'll want to check `resp.ok` before processing the payload. Even a bad request (400) would resolve successfully and be parsed by `.then`. This could lead the UI to hang.

(I always wish there was a more elegant way; let me know if you find one)

https://www.tjvantoll.com/2015/09/13/fetch-and-errors/

Reply:
Error handling added.